client/state/test/utils.js

```diff
@@ -121,27 +121,27 @@ describe( 'utils', () => {
 	} );
 
 	describe( '#createReducer()', () => {
-		describe( 'only default behavior', () => {
+		describe( 'with null initial state and no handlers', () => {
 			beforeAll( () => {
-				reducer = createReducer();
+				reducer = createReducer( null, {} );
 			} );
 
 			test( 'should return a function', () => {
 				expect( reducer ).to.be.a.function;
 			} );
 
-			test( 'should return initial state when invalid action passed', () => {
-				const invalidAction = {};
+			test( 'should return initial state when hydration action passed', () => {
+				expect( reducer( undefined, { type: '@@calypso/INIT' } ) ).to.be.null;
```

Comment:
Do we need this test? Note that it's the same as ` 'should return initial state when unknown action type passed'`. '@@calypso/INIT' was chosen so we wouldn't collide with any real actions. See also:
https://github.com/Automattic/wp-calypso/pull/9991#discussion_r92054348

Reply:
Oh, that "return initial state when unknown action" test is just incorrectly named. I'm changing the description from "initial state" to "identical state" in https://github.com/Automattic/wp-calypso/pull/22471/commits/fbdb456c3edf01ce83ed6eba0dc118fa7cd44e21. There are three similar tests:
- test that `undefined` state is hydrated into initial state when hydration action (i.e., one not explicitly handled) is dispatched
- test that invalid action (no `type` field) is processed and returns identical action
- test that unknown action converts valid (non-`undefined`) state to identical state